packages/conform-react/helpers.ts

```diff
@@ -60,24 +70,37 @@ type InputOptions = BaseOptions &
 		| {
 				type: 'checkbox' | 'radio';
 				value?: string;
+				id?: string;
 		  }
 		| {
-				type?: Exclude<HTMLInputTypeAttribute, 'button' | 'submit' | 'hidden'>;
+				type?: Exclude<InputProps['type'], 'button' | 'submit' | 'hidden'>;
 				value?: never;
+				id?: never;
```

Comment:
If the idea is to provide a way for user to override the default id, I think we can support it with all input type